app/src/main/java/com/example/appmarvel/services/serviceimplementation/GetCharactersListServiceImpl.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.appmarvel.services.serviceimplementation
+
+import com.example.appmarvel.entity.Character
+import com.example.appmarvel.services.ServiceGenerator
+import com.example.appmarvel.services.api.MarvelApi
+import com.example.appmarvel.services.api.mapper.CharacterMapper
+import com.example.appmarvel.services.serviceinterface.GetCharactersListService
+import com.example.appmarvel.services.utils.Result
+
+class GetCharactersListServiceImpl : GetCharactersListService {
+    private val api = ServiceGenerator()
+    private val mapper = CharacterMapper()
+
+    override fun getCharactersList(): Result<List<Character>> {
+        try {
+            val callResponse = api.createService(MarvelApi::class.java).getCharactersList()
+            val response = callResponse.execute()
+            if (response.isSuccessful)
+                response.body()?.let {
+                    mapper.transform(it)
+                }?.let {
+                    return Result.Success(it)
+                }
+        } catch (e: Exception) {
+            return Result.Failure(e)
+        }
+        return Result.Failure(Exception(CHARACTERS_NOT_FOUND))
+    }
+
+    companion object {
+        private const val CHARACTERS_NOT_FOUND = "The characters were not found"
```

Comment:
If you will show this text on the screen, this should be in a string resource, never hardcoded here

Reply:
Done